dialogue-client-test-lib/src/main/java/com/palantir/dialogue/AbstractChannelTest.java

```diff
@@ -196,165 +192,122 @@ public void fillsQueryParameters() throws Exception {
                 .isInstanceOf(SafeIllegalArgumentException.class)
                 .hasMessage("Endpoint must have a request body: {method=PUT}");
     }
 
     @Test
-    public void delete_failsWhenBodyIsGiven() throws Exception {
+    public void delete_failsWhenBodyIsGiven() {
         endpoint.method = HttpMethod.DELETE;
         when(request.body()).thenReturn(Optional.of(body));
-        assertThatThrownBy(() -> channel.createCall(endpoint, request).execute(observer))
+        assertThatThrownBy(() -> channel.createCall(endpoint, request))
                 .isInstanceOf(SafeIllegalArgumentException.class)
                 .hasMessage("DELETE endpoints must not have a request body");
     }
 
     @Test
     public void getMethodYieldsGetHttpCall() throws InterruptedException {
-        channel.createCall(endpoint, request).execute(observer);
+        channel.createCall(endpoint, request);
         assertThat(server.takeRequest().getMethod()).isEqualTo("GET");
     }
 
     @Test
     public void postMethodYieldsPostHttpCall() throws InterruptedException {
         endpoint.method = HttpMethod.POST;
         when(request.body()).thenReturn(Optional.of(body));
-        channel.createCall(endpoint, request).execute(observer);
+        channel.createCall(endpoint, request);
         assertThat(server.takeRequest().getMethod()).isEqualTo("POST");
     }
 
     @Test
     public void putMethodYieldsPutHttpCall() throws InterruptedException {
         endpoint.method = HttpMethod.PUT;
         when(request.body()).thenReturn(Optional.of(body));
-        channel.createCall(endpoint, request).execute(observer);
+        channel.createCall(endpoint, request);
         assertThat(server.takeRequest().getMethod()).isEqualTo("PUT");
     }
 
     @Test
     public void deleteMethodYieldsDeleteHttpCall() throws InterruptedException {
         endpoint.method = HttpMethod.DELETE;
-        channel.createCall(endpoint, request).execute(observer);
+        channel.createCall(endpoint, request);
         assertThat(server.takeRequest().getMethod()).isEqualTo("DELETE");
     }
 
     @Test
     public void callObservesSuccessfulResponses() {
-        channel.createCall(endpoint, request).execute(observer);
-        verify(observer).success(any());
-        verify(observer, never()).failure(any());
-        verify(observer, never()).exception(any());
-    }
-
-    @Test
-    public void cannotExecuteCallTwice() {
-        Call call = channel.createCall(endpoint, request);
-        call.execute(observer);
-        assertThatThrownBy(() -> call.execute(observer))
-                .hasMessage("Error, this call was already executed");
-    }
-
-    @Test
-    public void canCancelNonExecutedCalls() {
-        assertThatCode(() -> channel.createCall(endpoint, request).cancel()).doesNotThrowAnyException();
+        ListenableFuture<Response> call = channel.createCall(endpoint, request);
+        Futures.addCallback(call, callback, MoreExecutors.directExecutor());
```

Comment:
this is a bit odd now; why not assert that the future completed successfully?

Reply:
done